java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -513,6 +514,72 @@ void testReadParquetFull() {
     }
   }
 
+  @Test
+  void testReadORC() {
+    assumeTrue(Cuda.isEnvCompatibleForTesting());
+    ORCOptions opts = ORCOptions.builder()
+            .includeColumn("float1")
+            .includeColumn("boolean1")
+            .includeColumn("byte1")
+            .build();
+    try (Table table = Table.readORC(opts, TEST_ORC_FILE)) {
+      long rows = table.getRowCount();
+      assertEquals(2, rows);
+      assertTableTypes(new DType[]{DType.FLOAT32, DType.BOOL8, DType.INT8}, table);
+    }
+  }
+
+  @Test
+  void testReadORCBuffer() throws IOException {
+    assumeTrue(Cuda.isEnvCompatibleForTesting());
+    ORCOptions opts = ORCOptions.builder()
+            .includeColumn("float1")
+            .includeColumn("boolean1")
+            .includeColumn("byte1")
+            .build();
+
+    byte[] buffer = new byte[(int) TEST_ORC_FILE.length() + 1024];
+    int bufferLen = 0;
+    try (FileInputStream in = new FileInputStream(TEST_ORC_FILE)) {
+      bufferLen = in.read(buffer);
+    }
+    try (Table table = Table.readORC(opts, buffer, 0, bufferLen)) {
+      long rows = table.getRowCount();
+      assertEquals(2, rows);
+      assertTableTypes(new DType[]{DType.FLOAT32, DType.BOOL8, DType.INT8}, table);
+    }
+  }
+
+  @Test
+  void testReadORCFull() {
+    assumeTrue(Cuda.isEnvCompatibleForTesting());
+    try (Table table = Table.readORC(TEST_ORC_FILE)) {
+      long rows = table.getRowCount();
+      assertEquals(2, rows);
+
+      DType[] expectedTypes = new DType[]{
+              DType.BOOL8,
+              DType.INT8,
+              DType.INT16,
+              DType.INT32,
+              DType.INT64,
+              DType.FLOAT32,
+              DType.FLOAT64,
+              DType.STRING,
+              DType.STRING,
+              DType.INT32,
+              DType.STRING,
+              DType.INT32,
+              DType.STRING,
+              DType.STRING,
+              DType.INT32,
+              DType.STRING
+      };
+
+      assertTableTypes(expectedTypes, table);
```

Comment:
Same comment about comparing expected table values rather than a few table attributes.

Reply:
The type assertion is already covered by `assertTablesAreEqual`.